### DISCUSSION TITLE
step dependent on another job

### DISCUSSION THREAD

**Discussion by cyberhck** [community/community#29327] [Actions]
Just a feature request it might not even be possible, but let's add this here for the sake of having a discussion to see if this helps or not.
Consider the following build.yaml workflow file:
```yaml
on: [push]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - run: docker-compose up -d
      - run: make build
  dockerize:
    name: dockerize
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - run: make docker_build_and_push
  release:
    name: semantic-release
    runs-on: ubuntu-latest
    needs: [dockerize, build]
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - uses: cycjimmy/semantic-release-action@v2
```

The only reason `release` needs to wait on `dockerize` and `build` is so that we don't release broken builds for example.

It'd be nice to be able to move the dependency from an entire job to a step. That way, I can do everything I need to do in a Job in parallel, and wait for the final step.

I've considered writing some actions to do the waiting as well, not sure if that's the approach I should be taking.

Basically, I'd remove `release.needs` and place it in the last like for uses semantic release action. That step would need docker and build, it'd wait till the other 2 are done.

An action to do that and calling the API every 5 seconds would also work in theory. But having event based is more optimal I'm guessing.

What do you guys think? Are you guys waiting for the entire job to do your last job, but the last job is a little big?

**Comment by airtower-luna** [comment]
Dependencies are only supported at the job level, so that's not possible.

> An action to do that and calling the API every 5 seconds would also work in theory. But having event based is more optimal I'm guessing.

Mind that the runner would have to already be "working" on the job for that. You'd end up wasting energy, and (for a private repository) billable minutes. :wink: 

If you want to speed up the workflow over all, I see one thing in your sample workflow: Check if really all the jobs have to get the whole repository history (with `fetch-depth: 0`). For creating release notes I see the point, but tests shouldn't need it. For a large repository that can make a lot of difference.

If you have more steps that aren't in the sample, you could check if parallelizing them in additional jobs makes sense. It's not always possible, but may be worth checking.

**Reply by cyberhck** [reply]
exactly, I understand the billable hours. I do set fetch-depth to 0 for build, dockerize and tests, I just made this workflow smaller since pasting my entire workflow would be too big and isn't needed any way :)

true, I could parallelize, but they kinda are the preparation for the actual thing that happens :smile:

**Reply by airtower-luna** [reply]
> true, I could parallelize, but they kinda are the preparation for the actual thing that happens

Yeah, there's always the question of whether the effort to split things and transfer results (artifacts etc.) is worth the gain. :sweat_smile: Well, you're the one who knows the actual workflow. :wink:
